examples/multi.rs

```diff
@@ -0,0 +1,74 @@
+use std::env;
+use std::time::Duration;
+
+use anyhow::Result;
+
+use curl::easy::Easy;
+use curl::multi::{EasyHandle, Message, Multi};
+
+const REQUEST_TIMEOUT: Duration = Duration::from_secs(10);
+
+fn perform_requests(projects: &Vec<String>) -> Result<Vec<Option<u32>>> {
+    fn create_request(multi: &Multi, url: &str, token: usize) -> Result<EasyHandle> {
+        let version = curl::Version::get();
+        let mut request = Easy::new();
+        request.url(url)?;
+        request.timeout(REQUEST_TIMEOUT)?;
+        request.useragent(&format!("curl/{}", version.version()))?;
+        request.write_function(|data| {
+            // You can save the data here.
+            Ok(data.len())
+        })?;
+
+        let mut handle = multi.add(request)?;
+        handle.set_token(token)?;
+        Ok(handle)
+    }
+
+    fn parse_response(handle: &mut EasyHandle, message: &Message) -> Option<u32> {
+        message.result()?.ok()?;
+        handle.response_code().ok()
+    }
+
+    let multi = Multi::new();
+    let mut handles: Vec<_> = projects
+        .iter()
+        .enumerate()
+        .map(|(token, url)| create_request(&multi, url, token).unwrap())
+        .collect();
+
+    let mut statuses: Vec<Option<u32>> = Vec::new();
+    statuses.resize(projects.len(), None);
+
+    let mut completed = 0;
+    while completed < handles.len() {
+        // Let curl multi perform as much transfer as possible on every
+        // attached "easy" handles.
+        multi.perform()?;
+
+        multi.messages(|message| {
+            let token = message.token().expect("failed to get the token");
+            assert!(token <= statuses.len());
+            statuses[token] = parse_response(&mut handles[token], &message);
+            completed += 1;
+        });
+
+        std::thread::sleep(Duration::from_millis(16));
```

Comment:
I don't think that this `std::thread::sleep` is correct, all blocking should happen through the `Multi` interface itself.

Reply:
The rational was that the curl multi interface is particularly adapted to combine in an application that already has an "update" that has to be called regularly. (e.g., every frames)

In this particular example it's indeed better to let the multi interface block as it will react to the io events if there is any.

Reply:
Sleeping is basically never the correct way to block, however, so this needs to be done by different means.